.github/workflows/main.yml

```diff
@@ -41,13 +41,22 @@ jobs:
     if: ${{ github.event_name != 'schedule' || github.repository_owner == 'BottlecapDave' }}
     name: Unit Tests
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11", "3.12", "3.13"]
     steps:
       - name: Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
       - name: Setup
         uses: ./.github/actions/setup
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
```

Comment:
`actions/setup-python` is overkill...

The images have numerous python versions installed anyway - https://github.com/actions/runner-images/blob/f469601/images/ubuntu/Ubuntu2404-Readme.md#python

`update-alternatives --set python /usr/bin/python${{ matrix.python-version }}` (with appropriate escaping) would be cleaner and simpler...

Reply:
Some `actions/cache` for `pip` is probably fruitful too.